common/http_test.go

```diff
@@ -0,0 +1,245 @@
+package common_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/tmlibs/common"
+)
+
+func TestWriteSuccess(t *testing.T) {
+	w := httptest.NewRecorder()
+	common.WriteSuccess(w, "foo")
+	assert.Equal(t, w.Code, 200, "should get a 200")
+}
+
+var blankErrResponse = new(common.ErrorResponse)
+
+func TestWriteError(t *testing.T) {
+	tests := [...]struct {
+		msg  string
+		code int
+	}{
+		0: {
+			msg:  "this is a message",
+			code: 419,
+		},
+	}
+
+	for i, tt := range tests {
+		w := httptest.NewRecorder()
+		msg := tt.msg
+
+		// First check without a defined code, should send back a 400
+		common.WriteError(w, errors.New(msg))
+		assert.Equal(t, w.Code, http.StatusBadRequest, fmt.Sprintf("#%d: should get a 400", i))
+		blob, err := ioutil.ReadAll(w.Body)
+		if err != nil {
+			assert.Failf(t, "expecting a successful ioutil.ReadAll", "#%d", i)
+			continue
+		}
+
+		recv := new(common.ErrorResponse)
+		if err := json.Unmarshal(blob, recv); err != nil {
+			assert.Failf(t, "expecting a successful json.Unmarshal", "#%d", i)
+			continue
+		}
+
+		assert.Equal(t, reflect.DeepEqual(recv, blankErrResponse), false, "expecting a non-blank error response")
+
+		// Now test with an error that's .HTTPCode() int conforming
+
+		// Reset w
+		w = httptest.NewRecorder()
+
+		common.WriteError(w, common.ErrorWithCode(errors.New("foo"), tt.code))
+		assert.Equal(t, w.Code, tt.code, fmt.Sprintf("#%d: got=%d want=%d", i, w.Code, tt.code))
```

Comment:
Nice, thanks for letting know. Yeah:
1) The signature for assert.Equal kinda tripped me out and I couldn't intuitively guess what it did
http://godoc.org/github.com/stretchr/testify/assert#Equal
<img width="734" alt="screen shot 2017-08-02 at 11 23 36 am" src="https://user-images.githubusercontent.com/4898263/28886143-9da2ee78-7775-11e7-80ad-406905106454.png">

VS

https://golang.org/pkg/fmt/#Printf
<img width="880" alt="screen shot 2017-08-02 at 11 27 04 am" src="https://user-images.githubusercontent.com/4898263/28886156-aae45efa-7775-11e7-9b90-f602b04bc2b0.png">

Notice the lack of: `format string, args ...interface{}` which in the case of assert.Equal are `msgAndArgs ...interface{}`

I'll update that in a sec.